celery/backends/redis.py

```diff
@@ -198,6 +204,24 @@ def __init__(self, host=None, port=None, db=None, password=None,
 
         if url:
             self.connparams = self._params_from_url(url, self.connparams)
+        
+        # If we've received SSL parameters via query string or the 
+        # redis_backend_use_ssl dict, check ssl_cert_reqs is valid. If set 
+        # via query string ssl_cert_reqs will be a string so convert it here
+        if ('connection_class' in self.connparams 
+            and self.connparams['connection_class'] == redis.SSLConnection):
+            ssl_cert_reqs = self.connparams.get('ssl_cert_reqs', 'MISSING')
+            if ssl_cert_reqs in [CERT_REQUIRED, 'CERT_REQUIRED']:
```

Comment:
Perhaps an alternative not using multiple `if` branches, not sure if it is clearer, but one can see the transformation immediately:

```python
ssl_cert_reqs_missing = 'MISSING'  # could also be a constant
ssl_string_to_constant = { 'CERT_REQUIRED': CERT_REQUIRED, 
'CERT_OPTIONAL': CERT_OPTIONAL, 
'CERT_NONE': CERT_NONE }
ssl_cert_reqs = ssl_string_to_constant.get(ssl_cert_reqs, ssl_cert_reqs)
if ssl_cert_reqs == ssl_cert_reqs_missing:
    raise ...
# Other warnings here ...
```

What do you think?

Reply:
I agree the above looks much cleaner  - I'm assuming line 5 should have `ssl_cert_reqs_missing` as the second param to `get`? e.g.
```python
ssl_cert_reqs = ssl_string_to_constant.get(ssl_cert_reqs, ssl_cert_reqs_missing)
```
However, the current code works when the value for `ssl_cert_reqs` is either in string form or provided as a constant. With the above, wouldn't we need a check to see whether the provided value is a string first? 

Happy to switch to this approach of you think it's preferable.

Reply:
You are right, sorry I missed that step:
```python
ssl_cert_reqs = self.connparams.get('ssl_cert_reqs', ssl_cert_reqs_missing)
ssl_cert_reqs = ssl_string_to_constant.get(ssl_cert_reqs, ssl_cert_reqs)
# instead of `if ssl_cert_reqs == ssl_cert_reqs_missing:`
if ssl_cert_reqs not in (CERT_NONE, CERT_REQUIRED, CERT_OPTIONAL):
    raise ...
```
An example taking the above step-by-step:
1. `ssl_cert_reqs` will be either `CERT_NONE`, `'CERT_NONE'` or `'MISSING'`.
2. `ssl_cert_reqs` will be converted to the constant if string: `'CERT_NONE'` becomes `CERT_NONE`. Otherwise the previous value is retained: variable value is already a constant or `'MISSING'` string.
3. We finally assert that the variable value is one of the valid constants.

How does that sound?

Reply:
Sounds good. This gives us the following block to carry out all the necessary checks.

```python
if ('connection_class' in self.connparams and
            self.connparams['connection_class'] == redis.SSLConnection):
        ssl_cert_reqs_missing = 'MISSING'
        ssl_string_to_constant = { 'CERT_REQUIRED': CERT_REQUIRED,
                                   'CERT_OPTIONAL': CERT_OPTIONAL,
                                   'CERT_NONE': CERT_NONE }
        ssl_cert_reqs = self.connparams.get('ssl_cert_reqs', ssl_cert_reqs_missing)
        ssl_cert_reqs = ssl_string_to_constant.get(ssl_cert_reqs, ssl_cert_reqs)
        if ssl_cert_reqs not in ssl_string_to_constant.values():
            raise ValueError(E_REDIS_SSL_CERT_REQS_MISSING_INVALID)

        if ssl_cert_reqs == CERT_OPTIONAL:
            logger.warning(W_REDIS_SSL_CERT_OPTIONAL)
        elif ssl_cert_reqs == CERT_NONE:
            logger.warning(W_REDIS_SSL_CERT_NONE)
        self.connparams['ssl_cert_reqs'] = ssl_cert_reqs
```
This passes the tests. If all looks ok, I'll commit.

Reply:
Update committed